### PR TITLE
west: Add fish shell completion

### LINKED ISSUE #607: Add shell completion for fish shell
Currently the following shells are supported:
- bash
- zsh

The fish auto completion documentation:
https://fishshell.com/docs/current/completions.html

### PR DESCRIPTION
**Pull request by Emplis**
Add west completion for fish shell.

Update others files to include the support for fish shell.

Fixes https://github.com/zephyrproject-rtos/west/issues/607

### CODE REVIEW

**Comment by carlescufi** [comment]
Hey thanks for this! I will take a look next week.

**Comment by Emplis** [comment]
Rebased to fix CI failure.

**Review by carlescufi — APPROVED** [review]
I tested this locally and worked great. I can't really review the fish code but I am +1 on this.

**Comment by Emplis** [comment]
> I tested this locally and worked great. I can't really review the fish code but I am +1 on this.

Thanks.
For reference, I tried to follow what is done with the completions shipped with fish and especially the one that use subcommands like west (e.g. https://github.com/fish-shell/fish-shell/blob/master/share/completions/dnf.fish)

File: scripts/west_commands/completion/west-completion.fish [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+end
+
+# return the list of available west commands
+function __zephyr_west_complete_help
+    set -l builtin_cmds "init" "create a west repository" \
+                        "update" "update projects described in west manifest" \
+                        "list" "print information about projects" \
+                        "manifest" "manage the west manifest" \
+                        "diff" '"git diff" for one or more projects' \
+                        "statue" '"git status" for one or more projects' \
```
Comment (mbolivar-ampere) [inline scripts/west_commands/completion/west-completion.fish:165]: There's a typo here; can you please check these?

**Review by mbolivar-ampere — APPROVED** [review]
LGTM from a quick skim; can you please send a follow-up fixing the typo after checking the commands and help?

**Comment by kartben** [comment]
Hi @Emplis! Would you be willing to submit a PR to update https://docs.zephyrproject.org/latest/develop/west/install.html#enabling-shell-completion? It looks a bit outdated and it would be great if it'd mention zsh and fish :)

**Pull request merged**